components/net/tests/fetch.rs

```diff
@@ -198,35 +201,30 @@ fn test_fetch_blob() {
 }
 
 #[test]
-fn test_fetch_file() {
+fn test_fetch_opaque_filtered_file() {
+    // Fetching a local file through the full fetch() method
+    // should return an opaque filtered response.
     let path = Path::new("../../resources/servo.css")
         .canonicalize()
         .unwrap();
-    let url = ServoUrl::from_file_path(path.clone()).unwrap();
+    let url = ServoUrl::from_file_path(path).unwrap();
     let origin = Origin::Origin(url.origin());
     let mut request = Request::new(url, Some(origin), None);
 
     let fetch_response = fetch(&mut request, None);
-    assert!(!fetch_response.is_network_error());
-    assert_eq!(fetch_response.headers.len(), 1);
-    let content_type: Mime = fetch_response
-        .headers
-        .typed_get::<ContentType>()
-        .unwrap()
-        .into();
-    assert_eq!(content_type, mime::TEXT_CSS);
-
+    assert_eq!(fetch_response.response_type, ResponseType::Opaque);
+    assert_eq!(fetch_response.headers.len(), 0);
     let resp_body = fetch_response.body.lock().unwrap();
-    let file = fs::read(path).unwrap();
-
-    match *resp_body {
-        ResponseBody::Done(ref val) => {
-            assert_eq!(val, &file);
-        },
-        _ => panic!(),
-    }
+    assert_eq!(*resp_body, ResponseBody::Empty);
 }
 
+// TODO (#25693): We need a test that can pierce the opaque filtering of
+// a file fetch to make sure the file contents are correctly loaded.
+// script::dom::htmlimageelement::ImageContext has an example of this piercing,
+// getting the FetchMetadata via a FetchResponseListener trait.
+// The fetch method currently implemented for these unit tests isn't
+// equipped to do it.
```

Comment:
We should be able to use `fetch_response.actual_response()` to test the unfiltered response properties.

Reply:
Thank you, that works and is much easier than what I was thinking (although a unit test for the way ImageContext does it might also be nice, since that's doing non-WPT-observable things!)